psydac/api/ast/linalg.py

```diff
@@ -106,7 +106,8 @@ def __hashable_new__(cls, ndim, block_shape, comm_id=None, **kwargs):
 
         # Generate random tag, unique for all processes in MPI communicator
         tag = random_string(8)
-        if comm is not None and comm.size>1:
+        parallel = True
+        if comm is not None and comm.size>1 and parallel:
```

Comment:
I don't understand why you set `parallel = True` before the if block, rather than inside it

Reply:
I will delete this part, I had a strange bug, and as a quick fix I did those changes, but I have fixed it properly

Reply:
it deleted in [0a922eb](https://github.com/pyccel/psydac/pull/248/commits/0a922ebe5711d1e83d5f158155d0b4ca99315bae)